### ISSUE TITLE
test: app-tap/fail_main failed on freebsd after commit 157a2d88

### ISSUE THREAD

**Issue by avtikhon** [tarantool/tarantool#4537]
Tarantool version:
Master

OS version:
Freebsd 12

Bug description:
```
[006] Test failed! Result content mismatch:
[006] --- app-tap/fail_main.result	Tue Oct  1 10:50:04 2019
[006] +++ app-tap/fail_main.reject	Wed Sep 18 08:52:37 2019
[006] @@ -1,3 +1,15 @@
[006]  TAP version 13
[006]  1..1
[006] -ok - main script error is handled gracefully
[006] +not ok - main script error is handled gracefully
[006] +  ---
[006] +  filename: /home/vagrant/tarantool/test/app-tap/fail_main.test.lua
[006] +  line: 0
[006] +  trace:
[006] +  - line: 0
[006] +    source: '@/home/vagrant/tarantool/test/app-tap/fail_main.test.lua'
[006] +    filename: /home/vagrant/tarantool/test/app-tap/fail_main.test.lua
[006] +    what: main
[006] +    namewhat: 
[006] +    src: /home/vagrant/tarantool/test/app-tap/fail_main.test.lua
[006] +  ...
[006] +# failed subtest: 1
[006] 
[006] Last 15 lines of Tarantool Log file [Instance "app_server"][/home/vagrant/tarantool/test/var/006_app-tap/fail_main.test.lua.tarantool.log]:
```
Steps to reproduce:
./test-run.py app-tap/fail_main.test.lua

Optional (but very desirable):
https://gitlab.com/tarantool/tarantool/-/jobs/308644460
* coredump
* backtrace
* netstat

**Comment by Gerold103** [comment]
It is not a regression I think, the test was introduced in the same commit.

**Comment by avtikhon** [comment]
Found that the issue is the same as previously seen in issue #4271 :
```
Data segment size exceeds process limit
```

Test expected result:
```
~/tarantool/src/tarantool /tmp/e8g6Fg/script.lua
LuajitError: /tmp/e8g6Fg/script.lua:1: Error in the main script
```
Test current result:
```
~/tarantool/src/tarantool -e "os.execute([[~/tarantool/src/tarantool /tmp/e8g6Fg/script.lua]]) ; os.exit(0)"
Data segment size exceeds process limit
```

Investigated and found that root cause of the problem may be because of the issue:
https://www.gitmemory.com/issue/LuaJIT/LuaJIT/332/494103183

**Comment by avtikhon** [comment]
Found the fixing isuue patch:
https://github.com/LuaJIT/LuaJIT/commit/b0ecc6dd65a0b40e1868f20719c4f7c4880dc32d

**Comment by igormunkin** [comment]
@avtikhon, the fix LGTM, since it's necessary to shrink data segment only for FreeBSD/x64 with disabled LJ_GC64 machinery. The rationale is described [here](http://lua-users.org/lists/lua-l/2010-10/msg00625.html).

@kyukhin, please apply the given patch to our trunk.

**Comment by Totktonada** [comment]
The issue with LuaJIT was fixed in 8749b3e1215b52879bd289376ad453cea1542d17, tests were enabled in e68665503266a18a1a1e7b64be901b2a40afc54a.
